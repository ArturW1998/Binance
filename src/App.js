import React, { Fragment } from 'react';
import { Provider } from 'react-redux';
import { BrowserRouter as Router, Route } from 'react-router-dom';
import { BASE_NAME } from './constants';
import store from './store';
import MarketPairs from './containers/MarketPairs';
import Header from './components/Header';

const App = () => (
  <Provider store={store}>
    <Router basename={BASE_NAME}>
      <Fragment>
        <Header />
        <main className="main">
          <section className="container-fluid">
            <div className="row">
              <div className="col-12">
                <Route exact path="/" component={MarketPairs} />
              </div>
            </div>
          </section>
        </main>
      </Fragment>
    </Router>
  </Provider>
);

export default App;
