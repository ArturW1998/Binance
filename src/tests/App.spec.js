import React from 'react';
import { shallow } from 'enzyme';
import App from '../App';

describe('App', () => {
  const app = shallow(<App />);

  it('renders properly', () => {
    expect(app.debug()).toMatchSnapshot();
  });

  it('renders <Header />', () => {
    expect(app.find('Header')).toHaveLength(1);
  });

  it('render only one <main />', () => {
    expect(app.find('main')).toHaveLength(1);
  });
});
