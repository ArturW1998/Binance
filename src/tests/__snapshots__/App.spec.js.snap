// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`App renders properly 1`] = `
"<Provider store={{...}}>
  <BrowserRouter basename=\\"/Binance/\\">
    <Header />
    <main className=\\"main\\" data-testid=\\"MainWrapper\\">
      <section className=\\"container-fluid\\">
        <div className=\\"row\\">
          <div className=\\"col-12\\">
            <Route exact={true} path=\\"/\\" component={{...}} />
          </div>
        </div>
      </section>
    </main>
  </BrowserRouter>
</Provider>"
`;
